### DISCUSSION TITLE
Images missing after moving my mastodon vm to a new server

### DISCUSSION THREAD

**Discussion by hagenbauer** [mastodon/mastodon#27671] [Q&A] [answered]
hi, I have moved my mastodon vm to a new server. DNS and Letsencrypt are changed to the new server. Nginx config should be the same since I installed the NGinx proxy via ansible automation.  
So in theory everything should be the same. 

But in reality the images are missing on the new servers. So I obviously made a mistake.
What is the right place to start debugging the error? I do not see any error in /var/log/mastodon/*.log or in the nginx proxy log. 

I guess there is some background process fetching the images but I dont know the name or where to look for logs.

**Comment by StrangeMinds1960** [comment]
How did you move Mastodon?  Did you use these instructions -

https://docs.joinmastodon.org/admin/migrating/

Also, were your images stored locally or on an S3 bucket or similar?

**Reply by hagenbauer** [reply]
I made a tar file of the vm, copied it over to the new hypervisor. changed dns and gave it a go. 

In the meantime I found out that the images are there on the server (stored locally)

`ls -ls system/cache/media_attachments/files/111/347/895/615/532/084/original/ac9f0e12fb9abf71.jpeg
2233 -rw-r--r--  1 mstdn  wheel  2275080 Nov  3 18:54 system/cache/media_attachments/files/111/347/895/615/532/084/original/ac9f0e12fb9abf71.jpeg
`

but the nginx proxy that  cant find them

`  "GET /system/cache/media_attachments/files/111/347/895/615/532/084/original/ac9f0e12fb9abf71.jpeg HTTP/2.0" 404 651 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0" 0.027
`

nginx config of proxy in a different vm is the same as the old one

**Reply by StrangeMinds1960** [reply]
Would the file ownership / permissions have changed because of the move?

**Reply by hagenbauer** [reply]
owner ship stayed within the vm. All files are owned by the user who is running the mastodon services (checked with "find . ! -user mastodon"

**Reply by hagenbauer** [reply]
is there any way to get debug information from the process who is responsible for the delivery of the images to the upstream proxy?

**Comment by hagenbauer** [comment, marked as answer]
Ok. the answer was 

`RAILS_SERVE_STATIC_FILES=true
`
in the env.production file

No idea how it  was working on the old server.  Since there was a nginx proxy serving all content it should not have worked there too.

**Comment by falavu** [comment]
Hallo Hagenbauer,
i have this Problem too, but i do not understand the solution. 
What means " RAILS_SERVE_STATIC_FILES=true".
I added the Line into file "env.production" at the end. but nothing happens (restarted the machine).
what should i do?
did you find something about it in a documentation?
Thank You very much for hints. 

my 1-Man-Instance (fedi.bcna.de): Debian 12 with Mastodon from Yunohost on a MacMini CoreDuo. it worked fine for two Months. The Problem appears after a hanging Update. 
(please sorry for my cruel english)
